### ISSUE TITLE
Make sure that we're not tracking commits of closed PRs

### ISSUE THREAD

**Issue by dscho** [gitgitgadget/gitgitgadget#157]
The `update-commit-mappings` command in the `Update GitGitGadget's PRs` job takes quite a long time. I suspect that it looks at commits of PRs that were merged long ago.

**Comment by webstech** [comment]
I verified the openPRs list in the options only has the open PRs (<60).  In `CIHelper::updateCommitMappings`, `commitsInPu` has well over 40k entries that will get searched a number of times.  The range-diff for pull #15 (the oldest one still open) took some time (20 sec) and returned over 4k entries to run through a reg-ex.  While the performance of my box may not match the azure server it does appear that some of the actions in `update-commit-mappings` could take some time.

**Comment by dscho** [comment]
Right, and I don't think that we necessarily need to monitor the commits of already-merged branches.

We will still want to map responses on the Git mailing list even to closed PRs, but we can do that relatively cheaply because of the way we look up Message-IDs (we create pseudo blobs for them and then look up their Git notes, if any). That is independent of the ever-growing list of commitsInPu. I think we'll need to find a way to cull that list regularly.

**Comment by dscho** [comment]
Still an issue. See e.g. https://dev.azure.com/gitgitgadget/git/_build/results?buildId=37528&view=logs&j=92a4755a-6173-591e-5c07-8dbe1165e937&t=05dc4a6b-2706-5ea6-77c8-e14f00112193

Probably should stop trying to map commits already in `master`...

**Comment by dscho** [comment]
It seems that with #231 merged, the "update commit mappings" step only takes about 3.5 minutes (see e.g. [here](https://dev.azure.com/gitgitgadget/git/_build/results?buildId=44261&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b)), down from ~21 minutes (see [the last run without #231 "branch-deployed"](https://dev.azure.com/gitgitgadget/git/_build/results?buildId=43393&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b)).

So I declare victory on this ticket.
